arches/app/views/graph.py

```diff
@@ -121,6 +121,44 @@ def post(self, request, graphid):
         graph = Graph.objects.filter(graphid=graphid).select_for_update().get()
         data = JSONDeserializer().deserialize(request.body)
 
+        # update nodegroups
+        nodegroup_ids_to_serialized_nodegroups = {}
+        for serialized_nodegroup in data["graph"]["nodegroups"]:
+            nodegroup_ids_to_serialized_nodegroups[
+                serialized_nodegroup["nodegroupid"]
+            ] = serialized_nodegroup
+
+        for nodegroup in models.NodeGroup.objects.filter(
+            nodegroupid__in=nodegroup_ids_to_serialized_nodegroups.keys()
+        ):
+            nodegroup.cardinality = nodegroup_ids_to_serialized_nodegroups[
+                str(nodegroup.nodegroupid)
+            ]["cardinality"]
+            nodegroup.save()
```

Comment:
I know I'm preaching to the choir, but it'd be great if we didn't have to save here as well as in the graph proxy model. As we discussed earlier, it would require some significant changes and therefore is something we want to refactor before the next feature release.